sdk/extensions/Microsoft.Extensions.Azure/src/Internal/ClientFactory.cs

```diff
@@ -141,13 +160,60 @@ internal static TokenCredential CreateCredential(IConfiguration configuration, T
                     throw new InvalidOperationException($"Unable to find a certificate with thumbprint '{certificate}'");
                 }
 
-                var credential = new ClientCertificateCredential(tenantId, clientId, certs[0], identityClientOptions);
+                ClientCertificateCredential credential;
+                if (additionallyAllowedTenantsList != null)
+                {
+                    var options = new ClientCertificateCredentialOptions();
+                    foreach (string tenant in additionallyAllowedTenantsList)
+                    {
+                        options.AdditionallyAllowedTenants.Add(tenant);
+                    }
+                    credential = new ClientCertificateCredential(tenantId, clientId, certs[0], options);
+                }
+                else
+                {
+                    credential = new ClientCertificateCredential(tenantId, clientId, certs[0]);
```

Comment:
Just FYI it's totally acceptable to pass `null` for the options parameter. It could clean up a lot of the if..else statements here. This applies to the other credential types as well.

Reply:
You could also pass an options bag that has no options set.

Reply:
Thanks, I think passing an empty options bag would be best.